code/datums/abilities/flock/flockmind.dm

```diff
@@ -467,12 +499,47 @@
 	if(HAS_ATOM_PROPERTY(target,PROP_ATOM_FLOCK_THING))
 		if (isflockdeconimmune(target)) // ghost structure on click opens tgui window
 			return TRUE
+		if (!src.tutorial_check(FLOCK_ACTION_MARK_DECONSTRUCT, target))
+			return TRUE
 		var/mob/living/intangible/flock/F = holder.owner
 		F.flock.toggleDeconstructionFlag(target)
 		return FALSE
 	return TRUE
 
+/datum/targetable/flockmindAbility/tutorial
+	name = "Interactive Tutorial"
+	desc = "Check out the interactive flock tutorial to get started."
+	icon_state = "question_mark"
+	cooldown = 0 SECONDS
+	targeted = FALSE
+
+/datum/targetable/flockmindAbility/tutorial/cast(atom/target)
+	if (..())
+		return TRUE
+	var/mob/living/intangible/flock/flockmind/flockmind = holder.owner
+	if (istype(flockmind) && flockmind.tutorial)
```

Comment:
```suggestion
	if (flockmind.tutorial)
```
Type checking needed? Not done in tutorial exit

Reply:
It's a good idea since this holder is used for both flockminds and flocktraces, and the reason it isn't in the exit proc is because if you've got into the tutorial then you're a flockmind.

Reply:
Tutorial button is only added to Flockminds so don't see how Flocktraces could get it?